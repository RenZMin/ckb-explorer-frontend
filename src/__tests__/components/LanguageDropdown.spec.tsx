import { ReactElement } from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import Language, { languageText } from '../../components/Dropdown/Language'
import i18n from '../../utils/i18n'

describe('Language Dropdown Component', () => {
  let component: ReactElement

  beforeAll(() => {
    component = <Language setShow={() => {}} top={0} left={0} />
  })

  it('shallow renders', () => {
    const wrapper = renderer.create(component).toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  it('Component Render', () => {
    const wrapper = shallow(component)
    expect(wrapper).toBeDefined()
    expect(wrapper.find('.language__selected')).toHaveLength(1)
    expect(wrapper.find('.language__normal')).toHaveLength(1)
  })

  it('languageText function', () => {
    expect(languageText('en')).toBe(i18n.t('navbar.language_en'))
    expect(languageText('zh', true)).toBe(i18n.t('navbar.language_en'))
  })
})
