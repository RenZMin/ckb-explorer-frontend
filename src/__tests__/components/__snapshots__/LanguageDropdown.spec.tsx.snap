// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Language Dropdown Component shallow renders 1`] = `
<div
  className="sc-bdnxRM fCNNGv"
  onMouseLeave={[Function]}
>
  <div
    className="sc-gtsrHT kcdbmG language__selected"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseOver={[Function]}
    role="button"
    tabIndex={-1}
  >
    简体中文
  </div>
  <div
    className="language__separate"
  />
  <div
    className="sc-gtsrHT kcdbmG language__normal"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseOver={[Function]}
    role="button"
    tabIndex={-1}
  >
    简体中文
  </div>
</div>
`;
